### PR TITLE
 Implement routes, controllers and middlewares

### PR DESCRIPTION
**Pull request by gcobcqwe**

### CODE REVIEW

File: server/controllers/points.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+/**
+ * @description
+ * @param {Request} req
+ * @param {Response} res
+ */
+async function generateCode( req, res ) {
+  try {
+    const uid = req.token.payload.sub;
+    const points = req.body.points;
+    const result = await pointsServiceInstance.generateCode(uid, points);
```
Comment (john0312) [inline server/controllers/points.js:41]: Maybe double check the type of points?

File: server/controllers/points.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+/**
+ * @description
+ * @param {Request} req
+ * @param {Response} res
+ */
+async function redeemCode( req, res ) {
+  try {
+    const uid = req.token.payload.sub;
+    const code = req.body.code;
```
Comment (john0312) [inline server/controllers/points.js:57]: Maybe double check the type of code?

File: server/controllers/points.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+ * @param {Request} req
+ * @param {Response} res
+ * @return {Promise<*>}
+ */
+async function transactions( req, res ) {
+  try {
+    const sender = req.token.payload.sub;
+    const receiver = req.body.receiver;
+    const points = req.body.points;
+    await pointsServiceInstance.transactions(sender, receiver, points);
```
Comment (john0312) [inline server/controllers/points.js:77]: Maybe double check the type that's coming from the user?

File: server/controllers/users.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    const result = await usersServiceInstance.findAll();
+    res.status(StatusCodes.OK).send(result);
+  } catch (e) {
+    logger.error(e);
+    res.status(StatusCodes.BAD_REQUEST).send({message: ReasonPhrases.BAD_REQUEST});
+  }
+}
+
+/**
+   * @description
```
Comment (john0312) [inline server/controllers/users.js:65]: Maybe add a comment on what this API is for and who can use it?

File: server/controllers/users.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    await usersServiceInstance.add(uid, role, points);
+    res.status(StatusCodes.OK).send({message: ReasonPhrases.OK});
+  } catch (e) {
+    logger.error(e);
+    res.status(StatusCodes.BAD_REQUEST).send({message: ReasonPhrases.BAD_REQUEST});
+  }
+}
+
+/**
+   * @description
```
Comment (john0312) [inline server/controllers/users.js:83]: Maybe add a comment on what this API is for and who can use it?

**Review by john0312 — APPROVED** [review]

File: server/ExpressLoader.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  /**
+   * @description Create an instance of ExpressLoader.
+   */
+  constructor() {
+    this.app = express();
+    // Parse body params and attache them to req.body
+    this.app.use(express.urlencoded({extended: true}));
+    this.app.use(express.json());
+    // Set CORS - Cross Origin Resource Sharing
+    this.app.use(cors(config.cors_options));
```
[the remark this answers is not in the document]
Comment (gcobcqwe) [reply server/ExpressLoader.js:50]: I assumed the online shop will use these APIs.

**Pull request merged**